geth/params/config.go

```diff
@@ -205,6 +215,9 @@ type NodeConfig struct {
 	// LogToStderr defines whether logged info should also be output to os.Stderr
 	LogToStderr bool
 
+	// UpstreamConfig extra config for providing upstream infura server.
+	UpstreamConfig *UpstreamRPCConfig `json:"UpstreamConfig"`
```

Comment:
Why is it a pointer? Although there're already many such examples, but this is why I think static struct would work out better:
1. Less boilerplate code as the default value is already valid and has `Enabled: false`.
1. Less probability to catch panic somewhere if the value is not set.
1. Easier to check for `Enabled`: no need to check on `nil`.
1. Less values in heap in the end.

What do you think?